test/cmd.js

```diff
@@ -1,30 +1,42 @@
 var Embark = require('../lib/index');
 var Cmd = require('../lib/cmd');
 
+var passingLines = function () {
+  var lines = [];
+  lines.push('Initializing Embark Template....');
+  lines.push('Installing packages.. this can take a few seconds');
+  lines.push('Init complete');
+  return lines;
+};
+
 describe('embark.Cmd', function () {
   var cmd = new Cmd(Embark);
+  var pl = passingLines();
+  var appname = 'deleteapp';
 
   describe('#new', function () {
-    it('it should not create an app without a name', function (done) {
-      cmd.newApp(undefined, function (output) {
+    this.timeout(0);
+    it('it should create an app with a `name` argument set', function (done) {
+      cmd.newApp(appname, function (output) {
         var lines = output.split('\n');
-        assert.equal(lines[0], 'please specify your app Name');
-        assert.equal(lines[1], 'e.g embark new MyApp');
-        assert.equal(lines[2], 'e.g embark new --help for more information');
+        console.log(lines);
+        assert.equal(lines[0], pl[0]);
+        assert.equal(lines[1], pl[1]);
+        assert.equal(lines[2], pl[2]);
+        assert.equal(lines[3], 'App ready at ./' + appname);
+        done();
       });
-      done();
     });
 
-    it('it should create an app with a name', function (done) {
-      var appname = 'deleteapp';
-      cmd.newApp(appname, function (output) {
+    it('it should prompt when given an empty app name', function (done) {
+      cmd.newApp(undefined, function (output) {
         var lines = output.split('\n');
-        assert.equal(lines[0], 'Initializing Embark Template....');
-        assert.equal(lines[1], 'Installing packages.. this can take a few seconds');
-        assert.equal(lines[2], 'Init complete');
-        assert.equal(lines[3], 'App ready at ./' + appname);
+        console.log(lines);
+        process.stdout.write(appname);
```

Comment:
That's helpful but the test still exits 0 when calling. Maybe because mocha is also running in tty and this part is sending sig 0 to it to terminate.